modules/core/04-channel/types/acknowledgement_test.go

```diff
@@ -68,18 +74,19 @@ func (suite TypesTestSuite) TestAcknowledgement() { //nolint:govet // this is a
 
 			err := tc.ack.ValidateBasic()
 
-			if tc.expPass {
+			if tc.expValidates {
 				suite.Require().NoError(err)
+
+				// expect all valid acks to be able to be marshaled
+				suite.NotPanics(func() {
+					bz := tc.ack.Acknowledgement()
+					suite.Require().NotNil(bz)
+					suite.Require().Equal(tc.expBytes, bz)
+				})
 			} else {
 				suite.Require().Error(err)
 			}
 
-			// expect all acks to be able to be marshaled
-			suite.NotPanics(func() {
-				bz := tc.ack.Acknowledgement()
-				suite.Require().NotNil(bz)
-			})
```

Comment:
I moved the check up to the `ValidateBasic` success case because I don't want to make any assumptions how invalid instances are marchalled. Probybly they should never be marshalled.